[otlp] Docs: Highlight common issues with OTLP --> Prometheus (#3629)
grafana/mimir: docs/sources/mimir/operators-guide/configure/configure-otel-collector.md

      processors: [...]
      exporters: [..., otlphttp]
```

## Format considerations

> We currently follow the official specification described here: [OTLP Metric points to Prometheus](https://opentelemetry.io/docs/reference/specification/compatibility/prometheus_and_openmetrics/#otlp-metric-points-to-prometheus).

Reviewer: ```suggestion
We follow the official [OTLP Metric points to Prometheus](https://opentelemetry.io/docs/reference/specification/compatibility/prometheus_and_openmetrics/#otlp-metric-points-to-prometheus) specification.
```
currently : we version docs, so "currently" is current as of this version and therefore redundant
link text : make it as WYSIWYG as possible. Read this aloud to hear how a screen reader might read it. :)

You might experience the following common issues:

> #### Dots (.) are converted to \_
> 
> Because Prometheus metrics do not support `.` and `-` characters in metric or label names, they get converted to `_` in Prometheus.
> 
> For example:
> 
> `requests.duration{http.status_code=500, cloud.region=us-central1}` in OTLP becomes `requests_duration{http_status_code=”500”, cloud_region=”us-central1”}` in Prometheus.
> 
> #### Resource attributes are added to the `target_info` metric.
> 
> For details, see the [OpenTelemetry Spec](https://opentelemetry.io/docs/reference/specification/compatibility/prometheus_and_openmetrics/#resource-attributes).
> 
> However, `<service.namespace>/<service.name>` or `<service.name>` (if the namespace is empty), is added as the label `job`, and `service.instance.id` is added as the label `instance` to every metric.

Reviewer: ```suggestion
* Dots (.) are converted to \_
   Prometheus metrics do not support `.` and `-` characters in metric or label names. Prometheus converts these characters to `_`. For example:

   `requests.duration{http.status_code=500, cloud.region=us-central1}` in OTLP becomes 
   `requests_duration{http_status_code=”500”, cloud_region=”us-central1”}` in Prometheus.

* Resource attributes are added to the `target_info` metric.

   For details, see the [OpenTelemetry Resource Attributes](https://opentelemetry.io/docs/reference/specification/compatibility/prometheus_and_openmetrics/#resource-attributes) specification.

  However, `<service.namespace>/<service.name>` or `<service.name>` (if the namespace is empty), is added as the label `job`, and `service.instance.id` is added as the label `instance` to every metric.
```
Is `instance` indeed intended to be verbatim?

Reviewer: I have not tested this locally, so make sure that the context about each point hangs under the respective list item.

Author: > Is instance indeed intended to be verbatim?

Yes

      processors: [...]
      exporters: [..., otlphttp]
```

## Format considerations

> We currently follow the official specification described here: [OTLP Metric points to Prometheus](https://opentelemetry.io/docs/reference/specification/metrics/data-model/#otlp-metric-points-to-prometheus).

Reviewer: This target does not exist, so I attempted to reword this.

> The following sections describe common issues you might experience.

Reviewer: ```suggestion
In doing so, you might experience the following common issues:
```

> #### Dots (.) are converted to _
> 
> Because Prometheus metrics don't support `.` and `-` in metric or label names, they are converted to `_` in Prometheus.
> 
> For example:
> 
> `requests.duration{http.status_code=500, cloud.region=us-central1}` in OTLP becomes `requests_duration{http_status_code=”500”, cloud_region=”us-central1”}` in Prometheus.

Reviewer: ```suggestion
- Dots (`.`) are converted to `_`. Because Prometheus metrics do not support `.` and `-` characters in metric or label names, they get converted to `_` in Prometheus.

  For example:
  `requests.duration{http.status_code=500, cloud.region=us-central1}` in OTLP
  `requests_duration{http_status_code=”500”, cloud_region=”us-central1”}` in Prometheus
```

Reviewer: The format needs to be test...

   
   `requests_duration{http_status_code=”500”, cloud_region=”us-central1”}` in Prometheus

* Resource attributes are added to the `target_info` metric.

  However, `<service.namespace>/<service.name>` or `<service.name>` (if the namespace is empty), is added as the label `job`, and `service.instance.id` is added as the label `instance` to every metric.

>    For details, see the [OpenTelemetry Resource Attributes](https://opentelemetry.io/docs/reference/specification/compatibility/prometheus_and_openmetrics/#resource-attributes) specification.

Reviewer: @gouthamve, I reordered these lines (124 - 126).

> #### Resource attributes are added to `target_info` metric
> 
> As described in the [Resource Attributes](https://opentelemetry.io/docs/reference/specification/metrics/data-model/#resource-attributes-1) specification, we add resource attributes to a special `target_info` metric.
> 
> However, `<service.namespace>/<service.name>` or `<service.name>` (if namespace is empty), is added as `job` label and
> `service.instance.id` is added as `instance` label to every metric.

Reviewer: ```suggestion
- Resource attributes are added to the `target_info` metric. Resource attributes get added to the `target_info` metric. For details, see the [Metrics Data Model](https://opentelemetry.io/docs/reference/specification/metrics/data-model).
  
  However, `<service.namespace>/<service.name>` or `<service.name>` (if the namespace is empty), is added as the label `job`, and `service.instance.id` is added as the label `instance` to every metric.
```
The target at `#resource-attributes-1` does not seem to exist. Reword this section to reflect that.
